modules/katello_content_view.py

```diff
@@ -120,35 +119,34 @@ def find_repository(self, name, product, organization):
         else:
             self._module.fail_json(msg="No Repository found for %s" % name)
 
+    def find_repositories(self, repositories, organization):
+        return map(lambda repository: self.find_repository(repository['name'], repository['product'], organization), repositories)
+
     def content_view(self, name, organization, repositories=[]):
         updated = False
-        org = self.find_organization(organization)
+        organization = self.find_organization(organization)
 
-        content_view = self._entities.ContentView(self._server, name=name, organization=org)
+        content_view = self._entities.ContentView(self._server, name=name, organization=organization)
         response = content_view.search()
 
-        content_view = response[0] if len(response) == 1 else None
-
-        if not content_view:
+        if len(response) == 1:
+            content_view = response[0]
+        elif len(response) == 0:
             content_view = content_view.create()
             updated = True
 
-        if len(repositories) > 0:
-            repos = []
+        repositories = self.find_repositories(repositories, organization)
 
-            for repository in repositories:
-                repository = self.find_repository(repository['name'], repository['product'], org.name)
-                found = [repo for repo in content_view.repository if repo.id == repository.id]
-
-                if not found:
-                    content_view.repository.append(repository)
-                    updated = True
-
-            if updated:
-                content_view.update(['repository'])
+        def get_ids(repository):
+            return repository.id
```

Comment:
Is this an inline function definition? Seems strange a bit to do it this way or need to do it this way.